Antidote/ActiveCallViewController.m

```diff
@@ -310,6 +394,64 @@ - (void)setupIncomingCallViewForFriend:(NSString *)nickname
     }];
 }
 
+#pragma mark - Call Menu
+
+- (void)callMenuButtonPressed
+{
+    if (self.tableViewOfPausedCalls.hidden) {
+        self.tableViewOfPausedCalls.hidden = NO;
+    }
+    else {
+        self.tableViewOfPausedCalls.hidden = YES;
+    }
+}
+
+#pragma mark - UITableViewDelegate
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    [self.dataSource activeCallController:self resumePausedCallSelectedAtIndex:indexPath];
+    self.tableViewOfPausedCalls.hidden = YES;
+}
+
+#pragma mark - UITableViewDataSource
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    PauseCallTableViewCell *cell = [self.tableViewOfPausedCalls dequeueReusableCellWithIdentifier:[PauseCallTableViewCell reuseIdentifier]];
+
+    NSString *nickName = [self.dataSource activeCallController:self
+                            pausedCallerNicknameForCallAtIndex             :indexPath];
+
+    NSTimeInterval callDuration = [self.dataSource activeCallController:self
+                                        pauseTimeDurationForCallAtIndex          :indexPath];
+
+    [cell setCallerNickname:nickName andCallDuration:callDuration];
+    cell.delegate = self;
+
+    return cell;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    NSInteger numberOfPausedCalls = [self.dataSource activeCallControllerNumberOfPausedCalls:self];
+
+    self.callMenuButton.hidden = (numberOfPausedCalls == 0);
+
+    return numberOfPausedCalls;
+}
+
+#pragma mark PauseCallTableViewCellDelegate
+
+- (void)pauseCallCellEndPausedCallButtonTapped:(PauseCallTableViewCell *)cell
+{
+    NSIndexPath *indexPath = [self.tableViewOfPausedCalls indexPathForCell:cell];
+
+    [self.dataSource activeCallController:self endPausedCallSelectedAtIndex:indexPath];
+
+    self.tableViewOfPausedCalls.hidden = YES;
```

Comment:
It would be better for user to hide paused calls only if there are no paused calls left. That way user will be able to dismiss several calls at one time.